processor/src/chiplets/mod.rs

```diff
@@ -348,26 +348,27 @@ impl Chiplets {
     ///
     /// If the specified address hasn't been previously written to, four ZERO elements are
     /// returned. This effectively implies that memory is initialized to ZERO.
-    pub fn read_mem(&mut self, ctx: u32, addr: Felt) -> Word {
+    pub fn read_mem(&mut self, ctx: u32, addr: Felt) -> Result<Word, ExecutionError> {
         // read the word from memory
-        let value = self.memory.read(ctx, addr, self.clk);
+        let value = self.memory.read(ctx, addr, self.clk)?;
 
         // send the memory read request to the bus
         let lookup = MemoryLookup::from_ints(MEMORY_READ_LABEL, ctx, addr, self.clk, value);
         self.bus.request_memory_operation(&[lookup], self.clk);
 
-        value
+        Ok(value)
     }
```

Comment:
Same comment as above here and for other functions in this file: should we just use `u32` for memory addresses everywhere?